voluncheer/profiles/test_password_reset.py

```diff
@@ -0,0 +1,100 @@
+from django.test import TestCase
+from django.urls import reverse
+from profiles.models import User
+from django.core import mail
+from django.utils.http import urlsafe_base64_encode
+from django.utils.encoding import force_bytes
+from django.contrib.auth.tokens import default_token_generator
+
+
+class PasswordResetTest(TestCase):
+    def setUp(self):
+        self.user = User.objects.create_user(
+            email="testuser@example.com",
+            password="password",
+            is_active="True",
+            type="2",
+        )
+        self.test_user = User.objects.get(email="testuser@example.com")
+        self.test_user.is_active = True
+        self.test_user.save()
+        self.uidb64 = urlsafe_base64_encode(force_bytes(self.test_user.pk))
+        self.token = default_token_generator.make_token(self.test_user)
+
+    def test_password_reset_form_view(self):
+        response = self.client.get(reverse("password_reset"))
+        self.assertEqual(response.status_code, 200)
+        self.assertTemplateUsed(
+            response, "registration/password_reset_form.html"
+        )  # noqa E501
+
+    def test_password_reset_form(self):
```

Comment:
I don't quite get why you need to test for status code 200 and then 302? Is there a redirection happening only when email specified?

Reply:
Yes, when email is specified this means that the user requested to send the reset password link to their email. The test_password_reset_form makes sure that the email is sent and that it contains the correct information inside. The next function we test to see that the link works for the user which is passed the uid and token arguments. That url redirects to the password reset page where they can enter their new password (and confirm their new password). The next function makes sure that once they enter that information, the page redirects to password_reset_complete, and that the password is updated in the database for the user. Finally, the last function tests that the password_reset_complete page is working and calling the correct template. Does this make sense?